### PR TITLE
Adding Server TypeScript Documentation Infrastructure

### PR DESCRIPTION
**Pull request by subhankar-panda**
This PR adds automatic TypeScript documentation generation using https://typedoc.org/

Here is the general idea:- 
1) We make a normal code change to tesc.events on the master branch of this repo.
2) We have a Circle CI agent (ci@tesc.ucsd.edu) that pulls the code off master, runs typedoc on it and deploys to [our new documentation site](https://ucsdtesc.github.io/Check-in/server/)

ci@tesc.ucsd.edu is added to GitHub as @tesc-ci 

Frontend document generation coming soon™

### CODE REVIEW

**Comment by subhankar-panda** [comment]
Please squash while merging :) I had to debug off CircleCI because my local instance of docker broke 🥇

**Comment by RedbackThomson** [comment]
Should we add a CNAME to the Github for something like "docs.tesc.events" and then maybe we could have "docs.tesc.events/client" and "docs.tesc.events/server"

File: .circleci/config.yml [hunk 1]
```diff
@@ -44,6 +44,41 @@ jobs:
...
+      - checkout
+      - run: 
+          name: Install NPM
+          command: npm install
+      - run:
+          name: Build Server Documentation
+          command: npm run gen-server-docs
+      - persist_to_workspace:
+          root: ~/
+          paths: ./project/docs/server
```
Comment (RedbackThomson) [inline .circleci/config.yml]: What does this step do?

File: .circleci/config.yml [hunk 2]
```diff
@@ -44,6 +44,41 @@ jobs:
...
+    steps:
+      - checkout 
+      - attach_workspace:
+          at: ~/
+      - run:
+          name: Install and configure deps
+          command: |
+            npm install gh-pages@2.0.1
+            git config user.email "ci@tesc.ucsd.edu"
+            git config user.name "ci"
```
Comment (RedbackThomson) [inline .circleci/config.yml]: Can we give the bot a bit nicer name and username? `ci` seems so basic lol

File: .circleci/config.yml (hunk 1 above)
Comment (subhankar-panda) [reply .circleci/config.yml]: Makes the docs/ directory "persisted" between runs of jobs - we need the `docs/` dir generated from `docs-build` in `docs-deploy`

File: .circleci/config.yml (hunk 2 above)
Comment (subhankar-panda) [reply .circleci/config.yml]: sure lol

**Pull request merged**